samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,145 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <hal/video.h>
+#include <SDL.h>
+#include <SDL_ttf.h>
+#include <stdbool.h>
+
+const extern int SCREEN_WIDTH;
+const extern int SCREEN_HEIGHT;
+
+// SDL requires this symbol to exist
+void _exit() {
+}
+
+void main() {
+  int init_ret;
+  bool initialized_pbkit = false;
+  bool initialized_SDL   = false;
+  bool initialized_TTF   = false;
+
+  init_ret = pb_init();
+  if (init_ret != 0) {
+    debugPrint("pb_init() failed!");
+    XSleep(2000);
+    return;
+  }
+  initialized_pbkit = true;
+  pb_show_debug_screen();
+  
+  init_ret = SDL_VideoInit(NULL);
+  if (init_ret != 0) {
+    debugPrint("SDL_VideoInit failed: %s", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+  initialized_SDL = true;
+
+  init_ret = TTF_Init();
+  if (init_ret != 0) {
+    debugPrint("TTF_Init failed: %s", TTF_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+  initialized_TTF = true;
+
+  SDL_Window   *window;
+  SDL_Renderer *renderer;
+  SDL_Surface  *surface;
+  SDL_Texture  *texture;
+  TTF_Font     *font;
+
+  window = SDL_CreateWindow("NXDK SDL_ttf sample",
+                            SDL_WINDOWPOS_UNDEFINED,
+                            SDL_WINDOWPOS_UNDEFINED,
+                            SCREEN_WIDTH, SCREEN_HEIGHT,
+                            SDL_WINDOW_SHOWN);
+  if (window == NULL) {
+    debugPrint("Window creation failed: %s", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  renderer = SDL_CreateRenderer(window, -1, 0);
+  if (renderer == NULL) {
+    debugPrint("CreateRenderer failed: %s\n", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  font = TTF_OpenFont("vegur-regular.ttf", 96);
+  if (font == NULL) {
+    debugPrint("Couldn't load font: %s", TTF_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  SDL_Color textcolor = {0x7F, 0xFF, 0x7F, 0xFF};
+  surface = TTF_RenderText_Blended(font, "NXDK", textcolor);
+  TTF_CloseFont(font);
+  if (surface == NULL) {
+    debugPrint("TTF_RenderText failed: %s", TTF_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  texture = SDL_CreateTextureFromSurface(renderer, surface);
+  SDL_FreeSurface(surface);
+  if (texture == NULL) {
+    debugPrint("Couldn't create texture: %s\n", SDL_GetError());
+    XSleep(2000);
+    goto cleanup;
+    return;
+  }
+
+  SDL_Rect outputPosition;
+  int window_height, window_width;
+  SDL_GetWindowSize(window, &window_width, &window_height);
+  SDL_QueryTexture(texture, NULL, NULL, &outputPosition.w, &outputPosition.h);
+  outputPosition.x = window_width/2 - outputPosition.w / 2;
+  outputPosition.y = window_height/2 - outputPosition.h / 2;
+
+  int done;
+  SDL_Event event;
```

Comment:
Can be moved into the loop where it's actually used (near event check - so the reader sees which types are used).

Reply:
I'd prefer not to re-initialize it on every loop though?

Reply:
It's not re-initialized (a variable declaration without initialization will mostly be a compile-time operation). The code will likely remain unchanged (variable will have reserved room on stack) - it's just easier to find for readers of the code (and it might be overwritten by other variables if it's out-of-scope; so there's less memory use).